api/controller/assignments.js

```diff
@@ -0,0 +1,321 @@
+                            uniqueJournalIds.push(stds[i].private_journal.toString());
+                        }
+                    }
+
+                    privateJournalIds = uniqueJournalIds;
+                    var entryDoc = {};
+
+                    db.collection(journalCollection, function (err, collection) {
+                        if (err) {
+                            res.send({ "error": "An error has occurred in finding journal Entry" });
+                        } else {
+                            //find entry by private_journal and ObjectID
+                            console.log(assignmentData.assignedWork.toString());
+                            //aggregate query to get entry on the bisis of private_journal and objectId
+                            collection.aggregate([
+                                {
+                                    $match: {
+                                        '_id': new mongo.ObjectID(req.user.private_journal),
+                                        "content.objectId": assignmentData.assignedWork
+                                    }
+                                },
+                                {
+                                    $unwind: "$content"
+                                },
+                                {
+                                    $match: {
+                                        "content.objectId": assignmentData.assignedWork
+                                    }
+                                }
+                            ]).toArray(function (err, entry) {
+                                entryDoc = entry[0].content;
+                                console.log(entryDoc);
+                                // adding assignment metadata to entry
+                                entryDoc.metadata.assignmentId = req.params.assingid;
+                                entryDoc.metadata.submissionDate = null;
+                                entryDoc.metadata.isSubmitted = false;
+
+                                for (var i = 0; i < privateJournalIds.length; i++) {
+                                    entryDoc = createEntry(entryDoc);
+                                    console.log(entryDoc);
+                                    updateEntry(entryDoc, privateJournalIds[i], res).then(function (result) {
+                                        //send data to client
+                                        res.send({
+                                            'result': result,
+                                            'entry': entryDoc,
+                                        });
+                                    }).catch(function (err) {
+                                        res.send({
+                                            "error": err
+                                        });
+                                    });
+                                }
```

Comment:
You need to keep a counter of the entries that are updated and return the response only once when all the entries are updated.

As of now, this section of code won't work as expected and will lead to bugs. It needs to be fixed.